### PR TITLE
Hoist block offset multiplication to top of the loops

### PR DESCRIPTION
**Pull request by softins**
**Short description of changes**

Improve buffer offset calculation to avoid multiplication overflow warnings and improve efficiency

**Context: Fixes an issue?**

#2292 did not fix the CodeQL warnings, so this is an alternative and better approach.

**Does this change need documentation? What needs to be documented and how?**

No

**Status of this Pull Request**

Ready and should work.

**What is missing until this pull request can be merged?**

~Smoke test with a server or client.~ Client built on Pi works fine.

## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [x] I tested my code and it does what I want
- [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I've filled all the content above

### CODE REVIEW

**Review by pljones — APPROVED** [review]
It looks nicer than what was there...

File: src/server.cpp [hunk 1]
```diff
@@ -1354,10 +1356,12 @@ void CServer::MixEncodeTransmitData ( const int iChanCnt, const int iNumClients
 opus_custom_encoder_ctl ( pCurOpusEncoder, OPUS_SET_BITRATE ( CalcBitRateBitsPerSecFromCodedBytes ( iCeltNumCodedBytes, iClientFrameSizeSamples ) ) );
             // clang-format on
 
-            for ( size_t iB = 0; iB < (size_t) vecNumFrameSizeConvBlocks[iChanCnt]; iB++ )
+            for ( int iB = 0; iB < vecNumFrameSizeConvBlocks[iChanCnt]; iB++ )
```
Comment (ann0see) [inline src/server.cpp:1359]: int is hopefully big enough?

**Review by ann0see — COMMENTED** [review]
See below.

File: src/server.cpp (hunk 1 above)
Comment (pljones) [reply src/server.cpp:1359]: `int` is far too big :).

Comment (ann0see) [reply src/server.cpp:1359]: Ok.

**Review by ann0see — APPROVED** [review]

**Pull request merged**